### PR TITLE
Remove Registration of Factory Configuration object even with valid config

### PR DESCRIPTION
**Pull request by tcormackMW**
Do not allow factory components to be registered. Implemented by always marking Configurations as unsatisfied

### CODE REVIEW

**Review by pelliott-mathworks — APPROVED** [review]

File: compendium/DeclarativeServices/test/gtest/TestUpdateConfiguration.cpp [hunk 1]
```diff
@@ -237,14 +237,59 @@ namespace test
...
+        auto uniqueVal1 = 6;
+        auto configuration1 = configAdminService->GetConfiguration(configName + "~1");
+        cppmicroservices::AnyMap props1 { cppmicroservices::AnyMap::UNORDERED_MAP_CASEINSENSITIVE_KEYS,
+                                          { { "uniqueKey", uniqueVal1 } } };
+
+        auto fut1 = configuration1->Update(props1);
+        fut1.wait();
+        auto service0 = GetInstance<test::CAInterface>();
+        ASSERT_TRUE(service0)
+            << "Factory component should not be registered even with config matching exactly (with no ~)";
```
Comment (Burgch) [inline compendium/DeclarativeServices/test/gtest/TestUpdateConfiguration.cpp:288]: Update comment?

**Review by Burgch — APPROVED** [review]
Looks great, thanks @tcormackMW!

File: compendium/DeclarativeServices/src/manager/ConfigurationManager.cpp [hunk 2]
```diff
@@ -163,7 +163,7 @@ namespace cppmicroservices
         {
             bool allConfigsAvailable = configProperties.size() >= metadata->configurationPids.size();
 
-            if ((metadata->configurationPolicy != CONFIG_POLICY_REQUIRE) || (allConfigsAvailable))
+            if (((metadata->configurationPolicy != CONFIG_POLICY_REQUIRE) || (allConfigsAvailable)) && metadata->factoryComponentID.empty())
```
Comment (jeffdiclemente) [inline compendium/DeclarativeServices/src/manager/ConfigurationManager.cpp:166]: there should be a comment explaining why we don't register

**Review by jeffdiclemente — APPROVED** [review]

File: compendium/DeclarativeServices/test/gtest/TestUpdateConfiguration.cpp (hunk 1 above)
Comment (tcormackMW) [reply compendium/DeclarativeServices/test/gtest/TestUpdateConfiguration.cpp:288]: fixed

File: compendium/DeclarativeServices/src/manager/ConfigurationManager.cpp (hunk 2 above)
Comment (tcormackMW) [reply compendium/DeclarativeServices/src/manager/ConfigurationManager.cpp]: resolved

**Pull request merged**